modules/s3_bucket_subscription/main.tf

```diff
@@ -1,22 +1,20 @@
 locals {
   role_name           = regex(".*role/(?P<role_name>.*)$", var.lambda.role)["role_name"]
   statement_id_prefix = var.statement_id_prefix != "" ? var.statement_id_prefix : var.iam_name_prefix
-  bucket_arns         = var.bucket_arns
-  bucket_count        = length(var.bucket_arns)
 }
 
 resource "aws_lambda_permission" "allow_bucket" {
-  count               = local.bucket_count
+  for_each            = var.bucket_arns
   statement_id_prefix = local.statement_id_prefix
   action              = "lambda:InvokeFunction"
   function_name       = var.lambda.arn
   principal           = "s3.amazonaws.com"
-  source_arn          = local.bucket_arns[count.index]
+  source_arn          = each.key
 }
 
 resource "aws_s3_bucket_notification" "notification" {
-  count  = local.bucket_count
-  bucket = trimprefix(local.bucket_arns[count.index], "arn:aws:s3:::")
+  for_each = var.bucket_arns
+  bucket   = trimprefix(each.key, "arn:aws:s3:::")
```

Comment:
Prefer `data.aws_arn` for ARN parsing

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/arn

Reply:
addressed in https://github.com/observeinc/terraform-aws-lambda/pull/50